[BE] feat: 관리자 인증 구현 #146

yuyoungrhee opened the pull request:
<!-- PR 제목 예시
ex. [BE] 로그인 API 구현
ex. [AN] 페이지 구현 -->

## 📌 변경 내용 & 이유
<!-- ex. 
- 사용자 로그인 기능 구현
- 로그인 실패 시 에러 메시지 반환 처리 추가 -->
- 관리자 인증 처리
    - 관리자 로그인 기능 구현
    - admin 관련 api를 인증 후 접근할 수 있도록 보호

## 📸 스크린샷 (선택)
> UI 변경이 있을 경우 첨부
<img width="1494" height="826" alt="image" src="https://github.com/user-attachments/assets/a1f44436-84db-4ce2-8e88-0cf4f5abf3ac" />

## 🧪 테스트 방법 (선택)
<!-- 코드 리뷰하는 팀원이 돌려볼 수 있도록 작성해주기 
ex. ReservationTest의 XXXX 테스트메서드로 확인 
ex. 어떤화면의 어떤기능에서 확인 -->

테스트의 auth/config 패키지에 있는 테스트 돌려보시면 됩니다.
- `AdminSecurityConfigTest`
- `ApiSecurityConfigTest`

## 📢 논의하고 싶은 내용
<!-- 이러이러해서 이부분을 이러이러하게 구현했는데 이방식이 괜찮은지? -->
- 관리자 인증은 세션방식으로 구현했습니다.
    - 스프링에서 제공&권장하는 UserDetails와 UserDetailServices 이용해서 구현했습니다.
- PR 너무 거대해질 것 같아서 기존 기능에 E2E 테스트만 추가했습니다. application은 현재 간단한 CRUD만 존재해서 E2E로 커버 가능합니다. 나중에 따로 test 이슈 파서 진행하겠습니다.

## 🧩 관련 이슈
<!-- 이슈 태그: #123 -->
<!-- 이슈 닫기: close #123 -->
close #146 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **신규 기능**
  * 관리자 로그인 페이지 및 네비게이션 바가 추가되었습니다.
  * 관리자 인증 및 보안을 위한 Spring Security 설정이 도입되었습니다.
  * 관리자 인증 관련 UserDetailsService 구현이 추가되었습니다.
  * 카테고리, 키워드, 히어릿 관리에 대한 통합 테스트가 추가되었습니다.

* **버그 수정**
  * 관리 페이지 및 API에 CSRF 보호가 적용되어 보안이 강화되었습니다.
  * 리소스 미발견 시 404 응답 핸들링이 개선되었습니다.

* **리팩터링**
  * 관리자 비밀번호가 별도 값 객체로 안전하게 관리되도록 도메인 구조가 개선되었습니다.
  * 관리자 관련 엔드포인트의 반환값 및 응답 구조가 변경되어 응답 본문이 제거되었습니다.
  * 관리 페이지의 폼 제출 로직이 CSRF 토큰 헤더 포함 방식으로 통합 및 개선되었습니다.
  * API 보안 설정이 필터 체인 순서 및 적용 범위를 명확히 하도록 개선되었습니다.

* **문서화**
  * 관리자 관련 HTML 템플릿에 네비게이션 바 및 CSRF 메타 태그가 추가되고, 로그인 페이지 템플릿이 새로 작성되었습니다.

* **테스트**
  * 관리자 및 API 보안, 카테고리/키워드/히어릿 관리 기능에 대한 통합 테스트가 추가되었습니다.
  * 관리자 로그인 및 CSRF 토큰 취득을 돕는 테스트 헬퍼가 추가되었습니다.

* **자잘한 작업**
  * 테스트 데이터베이스 초기화 스크립트에 admin 테이블 정리가 추가되었습니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Byesol committed:
[BE] Merge branch 'develop-be' into 'feat/#146-admin-page-auth' #146

yuyoungrhee committed:
[BE] refactor: Password 객체를 통해 Admin 비밀번호 인코딩 #146

yuyoungrhee committed:
[BE] feat: 관리자 로그인 기능 및 설정 추가 #146

- AdminSecurityConfig 추가 및 Filter 설정
- 기존 SecurityConfig -> ApiSecurityConfig: 관리자 보안 설정과 restful API를 위한 보안 설정 분리

yuyoungrhee committed:
[BE] test: 필요없는 테스트 삭제 #146

yuyoungrhee committed:
[BE] feat: html의 form에 csrf 토큰 추가 #146

yuyoungrhee committed:
[BE] refactor: create시 의미없는 response body 제거 #146

yuyoungrhee committed:
[BE] test: admin 관련 security & E2E 테스트 추가 #146

yuyoungrhee committed:
[BE] feat: No Resource 관련 ExceptionHandler 추가 및 favicon.ico 설정 #146

yuyoungrhee committed:
[BE] test: 테스트에서  인증이 필요한 api 경로 수정 #146

- bookmarks 관련 api endpoint 변경으로 인한 테스트에서 사용하는 api도 변경 필요

yuyoungrhee committed:
[BE] fix: 관리자 페이지에서 로그인 성공시 항상 관리자 홈으로 이동하도록 수정 #146

- alwaysUse가 true면 이전 요청과 관계없이 항상 이 URL로 이동함

yuyoungrhee committed:
[BE] feat: 관리자 페이지 네비게이션바 추가 #146

yuyoungrhee committed:
[BE] refactor: UserDetails의 getUsername을 loginId로 변경 #146

- 로그인 시 사용하는 식별자이므로 닉네임이아니라 loginId를 사용하도록 변경

yuyoungrhee commented on backend/src/main/java/com/onair/hearit/admin/domain/AdminUserDetails.java:
```diff
@@ -0,0 +1,54 @@
+    @Override
+    public String getUsername() {
+        return admin.getNickname();
+    }
```
오 그러네요, 반영했어용

[[BE] refactor: UserDetails의 getUsername을 loginId로 변경](https://github.com/woowacourse-teams/2025-hEARit/pull/207/commits/398454249b94ea069fc125ea08b48384826dc9ec) https://github.com/woowacourse-teams/2025-hEARit/issues/146

yuyoungrhee commented on backend/src/main/java/com/onair/hearit/admin/domain/Password.java:
```diff
@@ -0,0 +1,24 @@
+@RequiredArgsConstructor(access = AccessLevel.PROTECTED)
+public class Password {
+
+    private static final BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();
```
Password는 현재 VO기 때문에 Spring Bean과 무관하게 동작하는 순수한 도메인 객체로 두고 싶습니다.
반영 안 할게여

yuyoungrhee committed:
[BE] refactor: 불필요한 출력문 제거 #146

yuyoungrhee commented on backend/src/main/java/com/onair/hearit/admin/domain/Password.java:
```diff
@@ -0,0 +1,24 @@
+
+    public Password(String rawPassword) {
+        this.value = encoder.encode(rawPassword);
+        System.out.println("value = " + value);
```
아이거 빼는 거 깜빡했다 뺄게요

yuyoungrhee committed:
[BE] refactor: rest api를 다루는 컨트롤러에서 @RestController 일관성 맞추기 #146

yuyoungrhee committed:
[BE] feat: 로그아웃 처리 및 필요없는 코드 정리 일관성 맞추기 #146

yuyoungrhee commented on backend/src/main/java/com/onair/hearit/admin/application/AdminCategoryService.java:
```diff
@@ -36,9 +36,8 @@ public List<CategoryInfoResponse> getAllCategories() {
         Category category = new Category(request.name(), request.colorCode());
-        Category saved = categoryRepository.save(category);
-        return CategoryInfoResponse.from(saved);
+        categoryRepository.save(category);
```
관리자 페이지에서는 생성 시 페이지 자체가 재로딩 되기 때문에 생성된 객체를 response로 줄 필요가 없습니다.

따라서 create하는 메서드의 경우 모두 void로 변경했습니다. (Hearit, Category, Keyword 모두 동일)

Byesol commented on backend/src/main/java/com/onair/hearit/admin/presentation/AdminCategoryController.java:
```diff
@@ -42,7 +42,7 @@ public ResponseEntity<List<CategoryInfoResponse>> readAllCategories() {
-        CategoryInfoResponse response = adminCategoryService.addCategory(request);
-        return ResponseEntity.created(URI.create("/")).body(response);
+        adminCategoryService.addCategory(request);
+        return ResponseEntity.created(URI.create("/")).build();
```
uri의 body에 더 이상 담을 필요가 없다면` ResponseEntity.created`를 쓰는 대신에 아래와 같이 코드를 작성하는 것도 좋아보여요

`return ResponseEntity.status(HttpStatus.CREATED).build();`

yuyoungrhee replied:
오... 이건 상의가 좀 필요해보이는데, 지금 안드랑 협업하고있어서 created시 전부 uri가 필요 없거든요. 바꾸려면 일관성 측면에서 한꺼번에 바꾸는 게 좋을듯요?!

Byesol commented on backend/src/main/java/com/onair/hearit/admin/domain/Admin.java:
```diff
@@ -23,15 +24,15 @@ public class Admin {
+    public Admin(String loginId, String rawPassword, String nickname) {
         this.loginId = loginId;
-        this.password = password;
+        this.password = new Password(rawPassword);
```
테스트를 위한 생성자지만 그래도 Password 객체를 주입 받는 방식이 객체 책임 분리 측면에서 더 나아보여요

yuyoungrhee replied:
Password는 Admin에 포함되어있는 VO기 때문에 내부에서 생성하도록 했습니다. Password를 생성하는 것까지 Admin의 책임이라고 생각합니다. Password라는 도메인이 굳이 외부에 노출될 필요는 없어보여요

Byesol replied:
인정합니다. 다시 생각해보니, 생성 책임을 외부로 돌릴 필요가 없겠네요

Byesol commented on backend/src/main/java/com/onair/hearit/auth/infrastructure/jwt/JwtAuthenticationFilter.java:
```diff
@@ -32,7 +32,7 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
 
         // 화이트리스트면 토큰이 없어도 그냥 통과
-        if (token == null && isWhitelisted(request)) {
+        if ((token == null || token.isBlank()) && isWhitelisted(request)) {
```
여기서 null이 있는데도 isBlank 검사 추가한 이유가 뭔가요?

yuyoungrhee replied:
안드로이드에서 Authorization 헤더에 token이 빈 값으로 들어가면 어떻게되냐고 물어봐서 혹시 몰라서 추가했습니다. 근데 아마 null로 오는 것 같긴 한데, 확실치 않아요. "Bearer " 이렇게만 올 시에 방어할 수 있습니다.

Byesol replied:
아하 이해했습니다

Byesol commented on backend/src/test/java/com/onair/hearit/admin/presentation/AdminCategoryControllerTest.java:
```diff
@@ -0,0 +1,88 @@
+        // then
+        assertThat(response.page()).isEqualTo(0);
+        assertThat(response.size()).isEqualTo(10);
+        assertThat(response.totalElements()).isEqualTo(20);
```
assertAll 로 묶는 게 좋아보여용

yuyoungrhee replied:
앗 빼먹었네요 추가했습니다.
[[BE] test: 여러 assert문이 있는 경우 assertAll 추가](https://github.com/woowacourse-teams/2025-hEARit/pull/207/commits/8126551ccc0c190825b7be8a6a8b709fc82aee7b) https://github.com/woowacourse-teams/2025-hEARit/issues/146

Byesol commented on backend/src/test/java/com/onair/hearit/admin/presentation/AdminKeywordControllerTest.java:
```diff
@@ -0,0 +1,88 @@
+        // then
+        assertThat(response.page()).isEqualTo(0);
+        assertThat(response.size()).isEqualTo(10);
+        assertThat(response.totalElements()).isEqualTo(20);
```
여기도 asseertAll로 묶는게 좋아보여요

yuyoungrhee replied:
추가했습니다!
[[BE] test: 여러 assert문이 있는 경우 assertAll 추가](https://github.com/woowacourse-teams/2025-hEARit/pull/207/commits/8126551ccc0c190825b7be8a6a8b709fc82aee7b) https://github.com/woowacourse-teams/2025-hEARit/issues/146

Byesol commented on backend/src/test/java/com/onair/hearit/admin/presentation/AdminSecurityTestHelper.java:
```diff
@@ -0,0 +1,45 @@
+        String newSessionId = loginResponse.getCookie("JSESSIONID");
+
+        //3.  로그인 성공 후 CSRF 재획득
+        Response afterLogin = RestAssured.given()
```
이 부분이 이해 안가는데 왜 위에서 CSRF 를 받았는데 여기서 다시 로그인 요청을 보내 CSRF를 획득해야하나요?

yuyoungrhee replied:
Spring Security 기본 설정에서 로그인 성공 시에는 세션 고정 공격(Session Fixation Attack) 방지를 위해 JSESSIONID를 새로 발급합니다.

그런데 로그인 시에도 csrfToken + sessionId가 필요하고, 로그인 후의 다른 api를 테스트할때도 csrfToken + sessionId가 필요해서 그렇습니다!

```
oldSessionId = EB40272D9655A06673A557042E28A7F1
csrfToken = mu28AvDh6A5v-ljZPG0zT1saOuRuu01j2OPtsl-C3A3FOhSrqt-NMMiH3TdCyW3qXkAHLTkiF9wLjH9O6tCLgD7nvjXxCSGZ

newSessionId = 6DFAA52C83F55719102B0F574B9310D5
newCsrf = onsKF7Nykiqp-CIGxuDnPExBEfahbZj49K-JkD9cO_GUbTinkEpsL4YR8BOEwRA19s3TXnknPM7CWvvVl8u9pQtsWMKmWVmX
```

로그를 찍어보면 이렇습니다...

Byesol replied:
친절한 설명 감사합니다!! 

```
        Response afterLogin = RestAssured.given()
                .cookie("JSESSIONID", newSessionId)
                .get("/admin/login");

        String newCsrf = Jsoup.parse(afterLogin.getBody().asString())
                .selectFirst("input[name=_csrf]")
                .attr("value");
```

그러면 위 부분은 로그인을 또 하는 게 아니라  로그인된 상태에서 /admin/login 페이지에 접근해서  CSRF 토큰을 다시 획득하는 과정인거죠?

yuyoungrhee replied:
정 확 합 니 다 get이라서 그냥 페이지에 접근하는 과정임요

Byesol reviewed (commented):
고생 많으셨어요!! 간단한 리뷰 남겼습니다.

gabean13 commented on backend/build.gradle:
```diff
@@ -28,13 +28,16 @@ dependencies {
     implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
     implementation 'org.springframework.boot:spring-boot-starter-validation'
     implementation 'org.springframework.boot:spring-boot-starter-thymeleaf'
+    implementation 'org.thymeleaf.extras:thymeleaf-extras-springsecurity6'
```
타임리프도 분리해서 쓰는거 어떤가요?

yuyoungrhee replied:
좋은 생각
[[BE] refactor: 타임리프 관련 의존성 순서 정리](https://github.com/woowacourse-teams/2025-hEARit/pull/207/commits/37cb99c98eb5baab68203426aa3a652c1031993e) https://github.com/woowacourse-teams/2025-hEARit/issues/146

gabean13 commented on backend/src/main/java/com/onair/hearit/admin/presentation/AdminKeywordController.java:
```diff
@@ -42,7 +42,7 @@ public ResponseEntity<List<KeywordInfoResponse>> readAllKeywords() {
     @PostMapping
     public ResponseEntity<KeywordInfoResponse> createKeyword(@RequestBody KeywordCreateRequest request) {
-        KeywordInfoResponse response = adminKeywordService.addKeyword(request);
-        return ResponseEntity.created(URI.create("/")).body(response);
+        adminKeywordService.addKeyword(request);
+        return ResponseEntity.created(URI.create("/")).build();
     }
```
지금 현재 이 메서드에서 KeywordInfoResponse는 사용되지 않는 건가요? body에 아무것도 안담기는 것 같아서요!

yuyoungrhee replied:
https://github.com/woowacourse-teams/2025-hEARit/pull/207#discussion_r2239360556

현재 클라이언트는 React 등 SPA가 아닌, 정적 HTML + JS 기반 서버 렌더링 구조입니다.
보통 Response로 보내면 SPA의 경우 페이지를 재로딩하지않고 컴포넌트만 갈아끼워서 업데이트한다든가 이런 식으로 활용되는데, 지금은 페이지를 전부 재로딩하고있어서 사실상 Response로 보내는 게 의미가 없습니다. 프론트에서 사용하고 있지 않아요!

고능하게 그렇게 가면 좋긴한데 우선 불필요한 정보를 보내지 않는 방향으로 리팩터링 했습니다. 나중에 구구의 조언에 따라서 react로 바꾸게 되면 다시 body에 response 실어서 내려줄 것 같긴 하네요

gabean13 commented on backend/src/main/java/com/onair/hearit/admin/presentation/AdminCategoryController.java:
```diff
@@ -42,7 +42,7 @@ public ResponseEntity<List<CategoryInfoResponse>> readAllCategories() {
     @PostMapping
     public ResponseEntity<CategoryInfoResponse> createCategory(@RequestBody CategoryCreateRequest request) {
-        CategoryInfoResponse response = adminCategoryService.addCategory(request);
-        return ResponseEntity.created(URI.create("/")).body(response);
+        adminCategoryService.addCategory(request);
+        return ResponseEntity.created(URI.create("/")).build();
     }
```
여기도 body에 아무것도 안담는데 Response<CategoryInfoResponse>가 있어서 헷갈리는 것 같아요. 그리고 create에 location header를 어떻게 할지 얘기해보면 좋을 것 같아용

yuyoungrhee replied:
create요청시 response를 아예 없애는 방향으로 리팩터링했는데 헷갈리나여?

실제 서비스 코드 현재 이렇습니다
``` java
    public void addCategory(CategoryCreateRequest request) {
        Category category = new Category(request.name(), request.colorCode());
        categoryRepository.save(category);
    }
```

yuyoungrhee replied:
아 잘못알아들었네요 고쳤어욧
[[BE] refactor: Controller에서 필요없는 Response 제거](https://github.com/woowacourse-teams/2025-hEARit/pull/207/commits/a8afb88bb9f2b077f211fd43602821f3e2e273f9) https://github.com/woowacourse-teams/2025-hEARit/issues/146

JO-eusan commented on backend/src/main/java/com/onair/hearit/admin/domain/AdminUserDetails.java:
```diff
@@ -0,0 +1,54 @@
+        this.admin = admin;
+    }
+
+    public Admin getAdmin() {
```
여기서 롬복을 안쓰신 이유가 있을까요?
다른 곳에서 사용하고 있다면 컨벤션을 맞추는 것도 좋아보여요!

yuyoungrhee replied:
어랏 왜이랬지 감사합니다 고쳤어용
[[BE] refactor: 롬복 적용 등 불필요한 코드 제거](https://github.com/woowacourse-teams/2025-hEARit/pull/207/commits/3f04c20bcf2d675ed04539933c1183e5decc64f1) https://github.com/woowacourse-teams/2025-hEARit/issues/146

gabean13 commented on backend/src/main/java/com/onair/hearit/auth/application/AdminUserDetailsService.java:
```diff
@@ -0,0 +1,26 @@
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String loginId) {
+        Admin admin = adminRepository.findByLoginId(loginId)
+                .orElseThrow(() -> new UsernameNotFoundException("존재하지 않는 관리자입니다."));
+        return new AdminUserDetails(admin);
+    }
+}
```
저희가 정의한 Exception이 아닌 다른 Exception을 던진 이유가 있나요?? 이건 제가 security를 잘 몰라서ㅠㅠ순수궁금증입니다!!
저는 개인적으로 저희가 정의한 exception이 아니라 다른 exception을 던지게 되면, 한두개는 괜찮지만 그런 경우가 쌓일수록 흐름이 분산돼서 에러 핸들링이 점점 내 손을 벗어난다..?라는 느낌이 있어서 지양하려고 했었거든용

yuyoungrhee replied:
AuthenticationManager는 내부적으로 UserDetailsService.loadUserByUsername()을 호출하고
해당 메서드에서 UsernameNotFoundException이 던져지면 인증 실패 처리를 합니다. 다른 예외가 오면 500 처리를 하게 돼여!

Spring Security 세션 기반 로그인 설계 시 표준 인터페이스인 UserDetails 기반으로 구현하고 있어요!
https://docs.spring.io/spring-security/reference/servlet/authentication/passwords/user-details-service.html

gabean13 replied:
아하 그렇군요 자세한 설명 감사합니당

yuyoungrhee committed:
[BE] test: 여러 assert문이 있는 경우 assertAll 추가 #146

JO-eusan commented on backend/src/main/java/com/onair/hearit/admin/domain/AdminUserDetails.java:
```diff
@@ -0,0 +1,54 @@
+    @Override
+    public boolean isEnabled() {
+        return true;
+    }
```
UserDetails 인터페이스에 디폴트로 반환값이 true로 되어 있는데 오버라이딩을 하신 이유가 있을까요?

yuyoungrhee replied:
굳이 할 필요 없었네요 생각없이 단축키 딸깍한 것 같습니다.. 😅
[[BE] refactor: 롬복 적용 등 불필요한 코드 제거](https://github.com/woowacourse-teams/2025-hEARit/pull/207/commits/3f04c20bcf2d675ed04539933c1183e5decc64f1) https://github.com/woowacourse-teams/2025-hEARit/issues/146

gabean13 commented on backend/src/main/java/com/onair/hearit/auth/config/AdminSecurityConfig.java:
```diff
@@ -0,0 +1,45 @@
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.web.SecurityFilterChain;
+
+@Order(1)
```
이 Order를 붙히신 기준이 궁금합니다(이것도 순수궁금증임)

yuyoungrhee replied:
Spring Security에서 필터링을 순서대로 하는데, 지금 구조상 반드시 Admin관련 Filter  -> Jwt Filter 순대로 걸러져야됩니다.

이유는:

AdminSecurityConfig: `/admin/**, /api/v1/admin/**`
ApiSecurityCofing: `/api/**`

이렇게 처리하고있어서 ApiSecurityConfig 필터가 먼저올 경우 /api/v1/admin/ 도 Jwt 검증에 걸려서 실패하게 됩니다.
조금 더 구체적인 거를 먼저 처리하고 전역적인 처리를 나중에 필터 걸어야하는 느낌쓰

gabean13 reviewed (commented):
고생하셨어용

JO-eusan commented on backend/src/main/java/com/onair/hearit/auth/application/AdminUserDetailsService.java:
```diff
@@ -0,0 +1,26 @@
+
+    private final AdminRepository adminRepository;
+
+    public AdminUserDetailsService(AdminRepository repo) {
```
파라미터 이름을 생략하기 보다는 다른 클래스들 처럼 풀네임을 적거나 롬복을 쓰는건 어떨까요?

yuyoungrhee replied:
롬복적용해서 제거했습니다.
[[BE] refactor: 롬복 적용 등 불필요한 코드 제거](https://github.com/woowacourse-teams/2025-hEARit/pull/207/commits/3f04c20bcf2d675ed04539933c1183e5decc64f1) https://github.com/woowacourse-teams/2025-hEARit/issues/146

JO-eusan commented on backend/src/main/java/com/onair/hearit/auth/presentation/AuthPageController.java:
```diff
@@ -0,0 +1,13 @@
+package com.onair.hearit.auth.presentation;
```
현재 auth/presentation 패키지에는 두 가지 컨트롤러가 있는데, 이 pageController는 관리자 웹 페이지 용이고 AuthController는 히어릿 앱에서 실제 사용하는 컨트롤러 인것 같습니다.
이때, 명명이 조금 애매해서 각 역할이 불분명하다고 생각해요. AdminAuthPageController와 같이 관리자를 명시해주거나 완전히 관리자 패키지를 이동하면 어떨까요?

yuyoungrhee replied:
인정합니다! 관리자패키지로 옮겼어요 -> AdminViewController
AuthPageController는 삭제했습니다

[[BE] refactor: admin 관련 view Controller를 auth에서 제거, AdminViewController로 이동](https://github.com/woowacourse-teams/2025-hEARit/pull/207/commits/c9725f0abb9da6f2429355a70222ee663d259d8a) https://github.com/woowacourse-teams/2025-hEARit/issues/146

JO-eusan reviewed (changes requested):
관리자 페이지 로그인 기능 확인했습니다! 
코드가 많아서 프로덕션 코드 위주로 리뷰 남겼어요.

yuyoungrhee committed:
[BE] refactor: 불필요한 Transactional 제거 #146

- 단일 save 시 트랜잭션 불필요

yuyoungrhee committed:
[BE] refactor: 롬복 적용 등 불필요한 코드 제거 #146

yuyoungrhee committed:
[BE] refactor: admin 관련 view Controller를 auth에서 제거, AdminViewController로 이동 #146

yuyoungrhee committed:
[BE] Merge branch 'develop-be' into feat/#146-admin-page-auth #146

# Conflicts:
#	backend/src/main/java/com/onair/hearit/auth/config/ApiSecurityConfig.java
#	backend/src/main/java/com/onair/hearit/common/exception/GlobalExceptionHandler.java

yuyoungrhee committed:
[BE] refactor: 타임리프 관련 의존성 순서 정리 #146

yuyoungrhee committed:
[BE] refactor: Controller에서 필요없는 Response 제거 #146

Byesol reviewed (approved):
리뷰 반영 확인해서 approve 할께요

gabean13 reviewed (approved):
고생하셨어용

JO-eusan reviewed (approved):
리뷰 확인했습니다. Approve 할게요!
